tests/test_bgp_neighbor_af.rb

```diff
@@ -89,57 +89,73 @@ def setup_ios_xr
     vpn   = 'vpnv4 unicast'
     evpn  = 'l2vpn evpn'
 
+    cfg = []
+
     # Global requirements
     # route policies
     str1 = 'route-map-in-name'
     str2 = 'route-map-out-name'
-    config("route-policy #{str1}", 'end-policy')
-    config("route-policy #{str2}", 'end-policy')
-    config("route-policy #{str1.reverse}", 'end-policy')
-    config("route-policy #{str2.reverse}", 'end-policy')
-    config('route-policy foo_bar', 'end-policy')
-    config('route-policy baz_inga', 'end-policy')
+    cfg << "route-policy #{str1}" << 'end-policy'
+    cfg << "route-policy #{str2}" << 'end-policy'
+    cfg << "route-policy #{str1.reverse}" << 'end-policy'
+    cfg << "route-policy #{str2.reverse}" << 'end-policy'
+    cfg << 'route-policy foo_bar' << 'end-policy'
+    cfg << 'route-policy baz_inga' << 'end-policy'
+    cfg << 'rd-set auto' << 'end-set'
+
+    # If any of the above config already exists, the output will contain
+    # a warning that matches our magic "warning regex" (due to the presence
+    # of a '%'), so just execute without checking for errors.
```

Comment:
I wouldn't say anything about 'our magic "warning regex"' - the presence of '%' is an IOS convention for warning/error messages. Perhaps just:

```ruby
    # If any of the above config already exists, we will get a
    # CLI warning message about replacing existing configuration.
    # Ignore these warnings, as they're intentional.
```